addons/web/static/src/views/relational_model.js

```diff
@@ -358,7 +358,10 @@ class DataPoint {
                 return value
                     ? value.map((property) => ({
                           ...property,
-                          value: this._parseServerValue(property, property.value ?? false),
+                          value:
+                            this.model.rootParams.viewMode !== "list" && ["date", "datetime"].includes(property.type)
+                                ? property.value
+                                : this._parseServerValue(property, property.value ?? false),
```

Comment:
I believe for consistency we should still expect a Date object in the relational model (so keep the actual code).

And fix the issue at the root:
https://github.com/odoo/odoo/blob/840ea11f14cba68a3d81e86ade4b3ff1410a6e02/addons/web/static/src/views/fields/properties/properties_field.js#L94-L98
```js
    get propertiesList() {
        const propertiesValues = (this.props.record.data[this.props.name] || []).map(property => {
            let value;
            if (["date", "datetime"].includes(property.type)) {
                // Date and Datetime are immutable luxon objects.
                value = property.value;
            }
            // Deep copy of the property.
            property = JSON.parse(JSON.stringify(property));
            // Recover the original Luxon object for date(time), because toJSON
            // luxon method does not return a compatible value.
            property.value = value || property.value;
            return property;
        });
        return propertiesValues.filter((definition) => !definition.definition_deleted);
    }
```

The issue is that doing a `JSON.stringify()` will call the Luxon.toJSON method on date(time) properties, which return a string and not a full representation of the Date object, so when we do `JSON.parse()` the returned object will have a `string` as the date value instead of the original `Luxon` object, and that string format is not compatible with `deserializeDate(Time)` (used by `_parseServerValue`)